### PR TITLE
Add se.principia_web.principia

### PR DESCRIPTION
**Pull request by jjwrap**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.

Physics based sandbox game.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:** I emailed the maintainer and they are ok with submission

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build se.principia_web.principia

File: se.principia_web.principia.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+app-id: se.principia_web.principia
+runtime: org.freedesktop.Platform
+runtime-version: "24.08"
+sdk: org.freedesktop.Sdk
+rename-appdata-file: se.principia_web.principia.metainfo.xml
```
Comment (hfiguiere) [inline se.principia_web.principia.yml:5]: this is the correct name. You shouldn't need to do it.

File: se.principia_web.principia.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+rename-desktop-file: principia.desktop
+rename-icon: principia
+command: principia
+copy-icon: true
+finish-args:
+  - "--socket=x11"
```
Comment (hfiguiere) [inline se.principia_web.principia.yml:11]: `x11` need `--share=ipc`

File: se.principia_web.principia.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  - "--socket=pulseaudio"
+  - "--device=all"
```
Comment (hfiguiere) [inline se.principia_web.principia.yml:13]: why `all` devices.

File: se.principia_web.principia.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  - "/include"
+  - "/lib/pkgconfig"
+  - "/share/aclocal"
+  - "/share/gtk-doc"
+  - "/share/man"
+  - "*.la"
+  - "*.a"
+
+modules:
+  - shared-modules/glu/glu-9.json
```
Comment (hfiguiere) [inline se.principia_web.principia.yml:25]: `shared-modules` git submodule is missing.

File: se.principia_web.principia.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+rename-appdata-file: se.principia_web.principia.metainfo.xml
+rename-desktop-file: principia.desktop
+rename-icon: principia
+command: principia
+copy-icon: true
+finish-args:
+  - "--socket=x11"
+  - "--socket=pulseaudio"
+  - "--device=all"
+  - "--share=network"
```
Comment (bbhtt) [inline se.principia_web.principia.yml:14]: why does it need network ?

Comment (jjwrap) [reply se.principia_web.principia.yml:13]: it connects to network to upload and play community levels

**Comment by jjwrap** [comment]
bot, build se.principia_web.principia

File: shared-modules/glu/glu-9.json
Comment (bbhtt) [inline shared-modules/glu/glu-9.json:1]: shared modules still isn't added properly.

Comment (bbhtt) [reply shared-modules/glu/glu-9.json:1]: try removing it and adding it back again https://github.com/flathub/shared-modules instructions on README.

Then just do `git add shared-modules/`

**Comment by rollerozxa** [comment]
> I emailed the maintainer and they are ok with submission

Where is this email you are talking about?

For the record I don't really mind either way if Principia gets a Flatpak package as long as I don't end up on the hook to maintain it. Whether it is a good idea to have someone whose upstream "contributions" amount to [this](https://github.com/Bithack/principia/issues/145) as the maintainer is another question, but that's for Flathub to decide.

**Comment by jjwrap** [comment]
> Where is this email you are talking about?

I ate it

**Pull request closed without merging**